### ISSUE TITLE
How to remove repository provider config

### ISSUE THREAD

**Issue by CharlesWeaveIO** [aws-amplify/amplify-hosting#2992]
Not sure how to fully remove and reset repository provider. Use case is first created provider for GitHub use of personal account, then the repository was moved to a corporate account that the same personal account has access to. However there is no way to get back to the initial setup of the provider to select the sub-accounts. Tried the amplify remove, amplify hosting  commands, no way to get back to initial repository setup.

![remove-aws-amplify-repository-provider](https://user-images.githubusercontent.com/104998037/190879755-767acfd9-a9c8-43f8-92be-2735e4dce2ea.png)

**Comment by hloriana** [comment]
Hi @CharlesWeaveIO 👋🏽 apologies for the delay. Amplify uses GitHub apps to authorize access to the repositories in your account. You can add fine grained access by selecting specific repositories and even repositories in GitHub organizations that you would like Amplify to access. In order to reset this configuration, you must either delete the GitHub app or revoke permissions for the GitHub app. Please follow these steps in GitHub:

1. Navigate to your profile -> settings.
2. Under **Integrations** select **Applications**
3. Under **Installed GitHub apps** search for the app associated with the AWS region you are currently working in. Click **Configure**
4. Scroll down to the **Danger zone** and select **Uninstall "AWS Amplify (aws-region)"**.

If this doesn't work then you may also have an Oauth app that needs to be disabled. In this case:

1. Navigate back to your GitHub applications
2. Under **Authorized Oauth Apps**, search for the app associated with your desired AWS region
3. Select the menu button next to it and click **Revoke**
4. Navigated to the AWS console and follow the steps to host a new app with a git provider. You should be prompted to authorize access to Github and the organizations should be listed if you have permissions to access them.


Please let us know if these steps unblock you. Thank you!

**Comment by CharlesWeaveIO** [comment]
Took both steps above, when going back into amplify add hosting it still picked up the original settings. No change from those steps at this time. I'll update if that changes.

**Comment by hloriana** [comment]
@CharlesWeaveIO when you followed the steps to connect Amplify to GitHub, did you see the organization listed as an option or were you only able to select your personal account?

**Comment by hloriana** [comment]
Also, please make sure you remove the Github app that corresponds to the AWS region you are trying to work in (ex. us-east-1).

**Comment by tonywei92** [comment]
@hloriana Mine is a bit similar, once I select a provider, for example GitHub, it's reverted to BitBucket (maybe previously we're using Bitbucket)

I attached the video here:

![ezgif-3-66b0afa0a0](https://user-images.githubusercontent.com/20355806/196327283-33fc8b45-787f-48e1-a66d-ce66291b9e26.gif)

**Comment by hloriana** [comment]
Hi @tonywei92 apologies for the delay. If you are still seeing this behavior, try to clear your browser data (cache) before switching to a different git provider. I was unable to reproduce this behavior on my end and was able to switch between git providers without issues. 

This seems to be slightly different from the original issue posted so please create a new issue if you are still experiencing this.

**Comment by hloriana** [comment]
I'm not able to reproduce the original issue in the description. I was able to successfully update GitHub app permissions to access a repository in another account's GitHub organization (once the owner approved the installation of the GitHub app).

Closing due to no response to previous comments: https://github.com/aws-amplify/amplify-hosting/issues/2992#issuecomment-1273720512
